### ISSUE TITLE
Maybe need to update the code 

### ISSUE THREAD

**Issue by YUHANG-Ma** [lucidrains/DALLE2-pytorch#143]
https://github.com/lucidrains/DALLE2-pytorch/blob/350a3d60456693a8ecdccc820e97dbb6b0c81866/train_decoder.py#L250

Hi, when I try to reload the model I trained before, I find there is a bug. I guess you could add ".dict()" after "**load_config" to avoid an error.

**Comment by lucidrains** [comment]
@YUHANG-Ma thanks for catching this!

**Comment by YUHANG-Ma** [comment]
Yes. But I met another issue. When I load the model for distributed training, I got this issue. Could you see how I can fix it?
![a46cb7cbef53cc5066b050e774360b9](https://user-images.githubusercontent.com/72591225/172434433-f0571ee0-c856-4024-baab-1fcd943093f0.png)

**Comment by lucidrains** [comment]
hmm, that's weird, `state_dict` should be there https://github.com/lucidrains/DALLE2-pytorch/blob/main/train_decoder.py#L214-L217 are you sure you didn't modify the original code?

**Comment by YUHANG-Ma** [comment]
![image](https://user-images.githubusercontent.com/72591225/172436220-a20b4e4b-18f5-480f-a039-9fa2d6875a27.png)
I added an if.

**Comment by lucidrains** [comment]
@YUHANG-Ma so move that conditional `hvd.rank() == 0` to this line https://github.com/lucidrains/DALLE2-pytorch/blob/main/train_decoder.py#L249

**Comment by YUHANG-Ma** [comment]
Okay thanks.

**Comment by lucidrains** [comment]
haha, i guess i shouldn't bother asking in the future. it is the characteristic sign that things are working when researcher's slink away to write their papers and i don't hear anything

**Comment by lucidrains** [comment]
@YUHANG-Ma we can always wait for https://github.com/lucidrains/DALLE2-pytorch/pull/121 to complete, which relies on huggingface's accelerate framework. it should be nearing maturity

**Comment by lucidrains** [comment]
@YUHANG-Ma in any case, i think the main issue was addressed! let me close the issue
